apps/condo/pages/auth/change-password.tsx

```diff
@@ -1,158 +1,172 @@
-        return runMutation({
-            mutation: changePassword,
-            variables: { data: { token, password, dv: 1, sender } },
-            onCompleted: async ({ data: { result } }) => {
-                await signInByPhone({
-                    phone: result.phone,
-                    password: form.getFieldValue('password'),
-                }, () => {
-                    auth.refetch().then(() => {
-                        setIsSaving(false)
-                        Router.push('/')
-                    })
-                })
-            },
-            intl,
-            form,
-        }).catch(() => {
-            setIsSaving(false)
-        })
-    }
-
-    const onResetPasswordClick = useCallback(() => Router.push('/auth/forgot'), [])
-
-    const [checkConfirmPhoneActionToken] = useLazyQuery(GET_PHONE_BY_CONFIRM_PHONE_TOKEN_QUERY, {
-        onError: error => {
-            setRecoveryTokenError(error)
-            setIsLoading(false)
-        },
-        onCompleted: () => {
-            setRecoveryTokenError(null)
+    }), [PleaseConfirmYourPasswordMessage, PleaseInputYourPasswordMessage, TwoPasswordDontMatchMessage, changeMessage, minPasswordLengthValidator, requiredValidator])
+
+    const authWithPhoneAndPassword = useCallback(async (phone: string, password: string): Promise<void> => {
+        try {
+            const sender = getClientSideSenderInfo()
+            const captcha = await executeCaptcha()
+            const res = await authenticateUserWithPhoneAndPasswordMutation({
+                variables: {
+                    data: {
+                        dv: 1,
+                        sender,
+                        captcha,
+                        password,
+                        phone,
+                        userType: UserType.Staff,
+                    },
+                },
+            })
+
+            if (res.errors || !res?.data?.result?.item?.id) {
+                const error: Error & { originalErrors? } = new Error('Authorization failed (request)')
+                error.originalErrors = res.errors
+                throw error
+            }
+        } catch (error) {
+            console.error('Authorization failed')
+            console.error(error)
+            throw error
+        }
+    }, [authenticateUserWithPhoneAndPasswordMutation, executeCaptcha])
+
+    const onSubmit = useCallback(async (values: typeof INITIAL_VALUES) => {
+        if (isLoading) return
+
+        setIsLoading(true)
+
+        try {
+            const { password } = values
+            const sender = getClientSideSenderInfo()
+            // const captcha = await executeCaptcha()
+
+            const res = await changePasswordWithTokenMutation({
+                variables: {
+                    data: {
+                        dv: 1,
+                        sender,
+                        password,
+                        token,
+                        // captcha,
```

Comment:
?

Reply:
This mutation does not have this field(
It will need to be added, but not in this pr